turtlebot2_demo/launch/turtlebot_joy.py

```diff
@@ -23,8 +23,12 @@ def launch(launch_descriptor, argv):
         exit_handler=restart_exit_handler,
     )
     ld.add_process(
+        cmd=['teleop_node'],
+        name='teleop_node',
+        exit_handler=restart_exit_handler,
+    )
+    ld.add_process(
         cmd=['joy_node'],
         name='joy_node',
-        # The joy node is required, die if it dies
-        exit_handler=default_exit_handler,
+        exit_handler=restart_exit_handler,
```

Comment:
I ran into a situation where use of restart_exit_handler in another launch file caused the turtlebot to lock up. I passed an invalid argument to map_server and it kept respawning and dying repeatedly. Perhaps we should be sparing with the use of the restart exit handler..!